assignments/this.js

```diff
@@ -1,26 +1,44 @@
 /* The for principles of "this";
 * in your own words. explain the four principle for the "this" keyword below.
 *
-* 1. 
-* 2. 
-* 3. 
-* 4. 
+* 1. Window binding, If we are in a gobal scope, or not bound to any specifically previoiusly adresed object this is bound to the window object in the broweser, or the console in node 
+* 2. Implicit binding, Whenever we call this while inside an specific object, this refers to the object. If we call it in a function of this object, the same applies.
+* 3. New Binding, If we call a constructor the this keyword refers to the new obect. we can use it for example to initialze its variables
+* 4. Explicit Binding, We can also set this explicitly by putting the object we want as this as the parameter of the apply or call method 
 *
```

Comment:
* 1. Ivoked with 'new' keyword - 'new' keyword makes a function call a constructor call. When a function is invoked with constructor call a new object is created (constructed) and set as 'this' binding for that function call.
* 2. Explicit binding - Most functions in JS have .call() and .apply() methods available to them. Both of these methods take an object as their first parameter on which 'this' should be used. This is how you explicitly say that you want 'this' to be the object passed in the parameter.
* 3. Implicit binding - A function can be declared outside the scope of an object.  If the object contains a reference to this function in its scope, and when calling the function 
it is preceded by the object reference (i.e.: obj.function()), than this object should be used for the functions 'this' binding (i.e.: this = obj). The object in implicit binding have to contain a reference in itself to the function!